### PR TITLE
[plotly.js] Added polar layout attributes

### PR DESCRIPTION
**Pull request by mtadams007**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://plot.ly/javascript/reference/#layout-polar>>

### CODE REVIEW

File: types/plotly.js/index.d.ts [hunk 1]
```diff
@@ -282,6 +282,12 @@ export interface Layout {
 	bargroupgap: number;
 	selectdirection: 'h' | 'v' | 'd' | 'any';
 	hiddenlabels: string[];
+	polar: {
```
Comment (szechyjs) [inline types/plotly.js/index.d.ts]: based on the docs there are many more properties than `bgcolor` and `radialaxis.color`. Could you make this feature complete? Something like:

```ts
polar: Partial<LayoutPolar>;
```

https://plot.ly/javascript/reference/#layout-polar

Also, as is, both of those properties would be required when polar is defined, as the `?` is missing on each.

**Review by szechyjs — CHANGES_REQUESTED** [review]

File: types/plotly.js/index.d.ts (hunk 1 above)
Comment (mtadams007) [reply types/plotly.js/index.d.ts]: Sure thing. Will get on it.

File: types/plotly.js/index.d.ts [hunk 2]
```diff
@@ -186,6 +186,80 @@ export interface DownloadImgopts {
...
+	domain: {
+		x: number[];
+		y: number[];
+		row: number;
+		column: number;
+	};
+	sector: number[];
+	hole: number;
+	bgcolor: Color;
+	radialaxis: {
```
Comment (szechyjs) [inline types/plotly.js/index.d.ts]: all of these nested properties should be partials...

```ts
radialaxis: Partial<{
  visible: boolean;
  range: any[];
}>;

Comment (mtadams007) [reply types/plotly.js/index.d.ts]: I made the radialaxis into a partial. I'm not super familiar with using Partials - should I make all the nested features within it also a partial (e.g. `title`)?

**Comment by mtadams007** [comment]
Closing in favor of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42832

**Pull request closed without merging**